core/lsp/QueryResponse.h

```diff
@@ -11,15 +11,16 @@ class TypeConstraint;
 class SendResponse final {
 public:
     SendResponse(core::Loc termLoc, std::shared_ptr<core::DispatchResult> dispatchResult, core::NameRef callerSideName,
-                 bool isPrivateOk, core::MethodRef enclosingMethod, core::Loc receiverLoc)
+                 bool isPrivateOk, core::MethodRef enclosingMethod, core::Loc receiverLoc, u2 totalArgs)
         : dispatchResult(std::move(dispatchResult)), callerSideName(callerSideName), termLoc(termLoc),
-          isPrivateOk(isPrivateOk), enclosingMethod(enclosingMethod), receiverLoc(receiverLoc){};
+          isPrivateOk(isPrivateOk), enclosingMethod(enclosingMethod), receiverLoc(receiverLoc), totalArgs(totalArgs){};
     const std::shared_ptr<core::DispatchResult> dispatchResult;
     const core::NameRef callerSideName;
     const core::Loc termLoc;
     const bool isPrivateOk;
     const core::MethodRef enclosingMethod;
     const core::Loc receiverLoc;
+    const u2 totalArgs;
```

Comment:
Why use a `u2` (which requires a static_cast)?

I don't think that this particular structure is performance sensitive w.r.t. allocation size.

Reply:
No special reason, I think I was following what some other underlying storage was somewhere else. I will do `size_t` instead to avoid casting.